### PR TITLE
upgrade to go1.18

### PR DESCRIPTION
**Pull request by ethan256**

### CODE REVIEW

File: .golangci.yml
Comment (wu-sheng) [inline .golangci.yml:127]: Disable? Because?

Comment (ethan256) [reply .golangci.yml:127]: Because structcheck is disabled in go1.18.  When using the latest version of golangci-lint, the following warning appears
```
make lint
...
WARN [linters context] structcheck is disabled because of go1.18. You can track the evolution of the go1.18 support by following the https://github.com/golangci/golangci-lint/issues/2649.
```

**Comment by wu-sheng** [comment]
@kezhenxu94 @mrproliu @fgksgf Wait for your review and evaluation of bumping up to 1.18.

**Review by kezhenxu94 — APPROVED** [review]
lgtm

**Comment by wu-sheng** [comment]
I think we may should do another release after merging this?

**Review by mrproliu — APPROVED** [review]
LGTM

**Comment by mrproliu** [comment]
> I think we may should do another release after merging this?

Sure. I could start to release it tonight.

**Pull request merged**
